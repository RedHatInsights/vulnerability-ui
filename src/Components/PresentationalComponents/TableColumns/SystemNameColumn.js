import React from 'react';
import PropType from 'prop-types';
import messages from '../../../Messages';
import { Label as Pflabel } from '@patternfly/react-core/dist/esm/components/Label/index';
import { intl } from '../../../Utilities/IntlProvider';
import { InsightsLink } from '@redhat-cloud-services/frontend-components/InsightsLink';

const SystemNameColumn = ({ item, system, optOut }) => {
    return (
        <span>
            {typeof item.title === 'undefined' ?
                <InsightsLink to={`/systems/${system.id}`}>{system.display_name}</InsightsLink>
                : item.title}
            { optOut && <Pflabel>{intl.formatMessage(messages.systemsTableDisabled)}</Pflabel>}
        </span>
    );
};

SystemNameColumn.propTypes = {
    item: PropType.object,
    system: PropType.object,
    optOut: PropType.bool
};

export default SystemNameColumn;
