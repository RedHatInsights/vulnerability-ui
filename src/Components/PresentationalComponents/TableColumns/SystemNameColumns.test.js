import React from 'react';
import toJson from 'enzyme-to-json';
import SystemNameColumn from './SystemNameColumn'

describe('SystemNameColumn', () => {
    it('Should match snapshot when not opt out', () => {
        const column = mount(<SystemNameColumn item={{ title: "Some system" }}/>)

        expect(toJson(column)).toMatchSnapshot();
    });

    it('Should match snapshot when not opt out', () => {
        const column = mount(<SystemNameColumn optOut item={{ title: "Some opt-out system" }}/>)

        expect(toJson(column)).toMatchSnapshot();
    });
})
