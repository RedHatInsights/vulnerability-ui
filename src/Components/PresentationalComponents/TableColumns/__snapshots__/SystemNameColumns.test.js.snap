// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemNameColumn Should match snapshot when not opt out 1`] = `
<SystemNameColumn
  item={
    {
      "title": "Some system",
    }
  }
>
  <span>
    Some system
  </span>
</SystemNameColumn>
`;

exports[`SystemNameColumn Should match snapshot when not opt out 2`] = `
<SystemNameColumn
  item={
    {
      "title": "Some opt-out system",
    }
  }
  optOut={true}
>
  <span>
    Some opt-out system
    <Label>
      <span
        className="pf-c-label"
      >
        <span
          className="pf-c-label__content"
        >
          Disabled
        </span>
      </span>
    </Label>
  </span>
</SystemNameColumn>
`;
